active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,311 @@
++       "deploymentId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Resources/deployments/my-deployment",
++       "alias": "az",
++       "name": "AzureResourceManager",
++       "version": "1.0.0"
++     },
++     {
++       "deploymentId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Resources/deployments/my-deployment",
++       "alias": "k8s",
++       "name": "Kubernetes",
++       "version": "1.27.8",
++       "config": {
++         // Out of the scope of this REP, but it is required to enable Deployment Stacks integration.  
++       }
++     }
+    ]
+  }
+}
+```
+
+#### Updating the `outputResources` to include more identifiable information
+
+To improve the current structure where each `outputResources` object only has a resource `id`, it's essential to add more properties for identifying extensible resources, which lack resource IDs. These additions should include the resource symbolic name, resource type, resource API version, resource identifier, the resource ID of the deployment containing the resource, and the Bicep extension alias.
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    "outputResources": [
+      // Azure resource
+      {
+        "id": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Storage/storageAccounts/my-storage-account"
+      },
+      // Extensible resource
+      {
++       "deploymentId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Resources/deployments/my-deployment",
++       "extensionAlias": "k8s",
++       "extension": "Kubernetes",
++       "symbolicName": "myService",
++       "resourceType": "core/Service",
++       "apiVersion": "v1",
++       "identifiers": {
++         "metadata": {
++           "namespace": "default",
++           "name": "myService",
++         },
++         "serverHostHash": "60fd32871cbe255d5793ce9e6ebf628beba25224cde7104e6a302f474f2f656e"
++       }
+      }
+    ]
+  }
+}
+```
+
+### API changes for `GET /Microsoft.Resources/deployments/operations/{operationId}` and `GET /Microsoft.Resources/deployments/operations`
+
+#### Updating `targetResource` to include extensible resource and extension information
+
+To align with modifications made to the `Microsoft.Resources/deployments` API, the response body now incorporates the addition of `symbolicName`, `resourceType`, `apiVersion`, `identifiers`, and `extension`.
+
+```diff
+{
+  ...
+  "properties": {
+    "provisioningOperation": "Create",
+    "provisioningState": "Succeeded",
+    "targetResource": {
+      "resourceType": "core/Service",
++     "apiVersion": "v1",
++     "symbolicName": "myService",
```

Comment:
Is `symbolicName` going to be added to the response for ARM resources?

Reply:
I'm not certain around the original points that led to that decision, but my guess is related to potential complications between non-symbolic and symbolic resource templates.

Reply:
Decision is to figure out if symbolic name is feasible to add now. It may be omitted in the implementation.

Reply:
I added this for at least the output resources in the new API changes. There's already metadata on the job for it so it should be easy to add to the operations as well.